File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1748.java
import java.util.*;

class Leetcode1748 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for(int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the arry: ", i+1);
            arr[i] = input.nextInt();
        }
        input.close();
        System.out.println(sumOfUnique(arr));
    }
    private static int sumOfUnique(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap();
        int sum = 0;
        for(int i: nums) {
            map.put(i, map.getOrDefault(i, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() == 1) {
                sum += entry.getKey();
            }
        }
        return sum;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1748.ts
function sumOfUnique(nums: number[]): number {
    const map = new Map<number, number>();
    let sum = 0;
    for (const num of nums) {
        map.set(num, (map.get(num) || 0) + 1); 
    }
    for (const [key, value] of map.entries()) {
        if (value === 1) {
            sum += key;
        }
    }
    return sum;
}

let nums: number[] = [1, 2, 3, 2];
console.log(sumOfUnique(nums));

export { };